Library/Unit/MonthUnit.cs

```diff
@@ -35,6 +44,14 @@ public MonthOnLastDayOfMonthUnit OnTheLastDay()
             return new MonthOnLastDayOfMonthUnit(Schedule, _duration);
         }
 
+        /// <summary>
+        /// Runs the job on the last week day of the month.
+        /// </summary>
+        public MonthOnLastWeekDayOfMonthUnit OnTheLastWeekDay()
+        {
+            return new MonthOnLastWeekDayOfMonthUnit(Schedule, _duration);
```

Comment:
Why another class? `MonthUnit` is designed to perform such job. Besides, you should give an instance of  `_calculator` to `MonthOnLastWeekDayOfMonthUnit` instead of `Schedule`.